parcels/particleset.py

```diff
@@ -238,38 +254,113 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
             self.remove(to_remove)
 
     def show(self, **kwargs):
-        if plt is None:
-            raise RuntimeError("Visualisation not possible: matplotlib not found!")
-
-        field = kwargs.get('field', True)
-        lon = [p.lon for p in self]
-        lat = [p.lat for p in self]
+        savefile = kwargs.get('savefile', None)
+        vector = kwargs.get('vector', False)
+        domain = kwargs.get('domain', None)
+        plon = np.array([p.lon for p in self])
+        plat = np.array([p.lat for p in self])
         time = [p.time for p in self]
-        t = int(kwargs.get('t', time[0]))
-        plt.ion()
-        plt.clf()
-        plt.plot(np.transpose(lon), np.transpose(lat), 'ko')
-        if field is True:
-            axes = plt.gca()
-            axes.set_xlim([self.grid.U.lon[0], self.grid.U.lon[-1]])
-            axes.set_ylim([self.grid.U.lat[0], self.grid.U.lat[-1]])
-            namestr = ''
-            time_origin = self.grid.U.time_origin
+        t = kwargs.get('t', time[0])
+        if isinstance(t, datetime):
+            t = (t - self.grid.U.time_origin).total_seconds()
+        if isinstance(t, delta):
+            t = t.total_seconds()
+        if domain is not None:
+            latN = nearest_index(self.grid.U.lat, domain[0])
+            latS = nearest_index(self.grid.U.lat, domain[1])
+            lonE = nearest_index(self.grid.U.lon, domain[2])
+            lonW = nearest_index(self.grid.U.lon, domain[3])
```

Comment:
Does this ordering of `domain` make most sense? I know that I suggested providing domain as a vector, but now I think perhaps better as separate arguments for `latS`, `lonE` etc. Especially if these default to the grid edges..